nat-lab/bin/core-api.py

```diff
@@ -132,31 +186,89 @@ def add_node(self, req):
             public_key=req.public_key,
             os=req.os,
             os_version=req.os_version,
-            device_type=req.device_type,
-            nickname=req.nickname,
-            discovery_key=req.discovery_key,
-            relay_address=req.relay_address,
+            device_type=req.device_type or "",
+            nickname=req.nickname or "",
+            discovery_key=req.discovery_key or "",
+            relay_address=req.relay_address or "",
             traffic_routing_supported=req.traffic_routing_supported,
         )
 
         self.server.add_machine(node)
 
-        pprint(node)
         return node
 
     def handle_machines_delete(self):
         uid = self.path.removeprefix(self.machines_path)[1:]
-        print("uuid:", uid)
 
-        try:
-            self.server.remove_machine(uid)
-            print(f"{uid} removed")
-            self.send_response(204)
-        except KeyError:
-            print(f"{uid} unknown")
-            self.send_response(400)
+        if self.server.remove_machine(uid):
+            self._set_headers(status_code=HTTPStatus.NO_CONTENT)
+        else:
+            self._send_error_response(
+                CoreApiErrorCode.MACHINE_NOT_FOUND,
+                "Machine not found",
+                HTTPStatus.NOT_FOUND,
+            )
+
+    def handle_patch_machine(self, machine_id):
+        machine = self.server.get_machines().get(machine_id)
+        if not machine:
+            self._send_error_response(
+                CoreApiErrorCode.MACHINE_NOT_FOUND,
+                "Machine not found",
+                HTTPStatus.NOT_FOUND,
+            )
+            return
 
-        self.end_headers()
+        content_length = int(self.headers["Content-Length"])
+        patch_data = self.rfile.read(content_length)
```

Comment:
Note that `Content-Length` header is not required, and I am not sure we include them in ESP32 and/or teliod implementations. Therefore I guess it would be better not to depend on it. One way to do that is:
```suggestion
        if "Content-Length" not in self.headers:
            patch_data = self.rfile.read()
        else:
            content_length = int(self.headers["Content-Length"])
            patch_data = self.rfile.read(content_length)
```

Reply:
I think Content-Length is required by HTTP/1.1 protocol which default protocol in [esp_http_client](esp_http_client https://github.com/espressif/esp-idf/blob/master/components/esp_http_client/esp_http_client.c#L161).
But when there is request being send with the payload, Content-Length is calculated by CURL (or esp_http_client in our case). 
For example executing below command:
>  curl -v -X POST http://localhost:8080/<endpoint> \
>    -H "Content-Type: application/json" \
>    -d '{"key": "test_value"
>        }'

Results to:
> POST /v1/meshnet/machines HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.88.1
> Accept: */*
> Content-Type: application/json
> **Content-Length: 238**